src/Components/Web.JS/src/Platform/Circuits/CircuitManager.ts

```diff
@@ -19,6 +20,9 @@ export class CircuitDescriptor {
       throw new Error('Circuit host not initialized.');
     }
 
+    if (reconnection.state !== HubConnectionState.Connected) {
+      return Promise.resolve(false);
+    }
     return reconnection.invoke<boolean>('ConnectCircuit', this.circuitId);
```

Comment:
does this throw immediately when is not connected?

Reply:
Technically it does a little processing of the args and internal setup for the invoke, but then it throws.

Reply:
Ok, then it's fine.

Just wanted to make sure that we weren't changing the behavior in case it internally tried to connect.